lib/NeuraDialect/Transforms/Optimizations/HwAgnosticOpt/FoldConstantPass.cpp

```diff
@@ -101,46 +101,67 @@ template <typename OpType>
 struct FuseRhsConstantPattern : public OpRewritePattern<OpType> {
   using OpRewritePattern<OpType>::OpRewritePattern;
 
+  // By default, we assume the operation is not commutative.
+  // If the operation is commutative, we can extend this pattern to support
+  // constant folding on the left-hand side operand as well.
+  virtual bool isCommutative() const { return false; }
+
   virtual Operation *
-  createOpWithFusedRhsConstant(OpType op, Attribute rhs_const_value,
+  createOpWithFusedRhsConstant(OpType op, Value non_const_operand,
+                               Attribute rhs_const_value,
                                PatternRewriter &rewriter) const = 0;
 
   LogicalResult matchAndRewrite(OpType op,
                                 PatternRewriter &rewriter) const override {
     Value lhs = op.getLhs();
     Value rhs = op.getRhs();
 
-    if (isOriginConstantOp(lhs)) {
-      llvm::errs() << "LHS constant folding not supported yet.\n";
-      return failure();
-    }
+    bool lhs_is_const = isOriginConstantOp(lhs);
+    bool rhs_is_const = rhs && isOriginConstantOp(rhs);
+
+    if (rhs_is_const && !lhs_is_const) {
+      auto constant_op = dyn_cast<neura::ConstantOp>(rhs.getDefiningOp());
 
-    if (!rhs || !isOriginConstantOp(rhs)) {
-      return failure();
+      Attribute rhs_const_value = getOriginConstantValue(rhs);
+      Operation *fused_op =
+          createOpWithFusedRhsConstant(op, lhs, rhs_const_value, rewriter);
+
+      rewriter.replaceOp(op, fused_op->getResults());
+      if (constant_op->use_empty()) {
+        rewriter.eraseOp(constant_op);
+      }
+      return success();
     }
 
-    auto constant_op = dyn_cast<neura::ConstantOp>(rhs.getDefiningOp());
+    if (lhs_is_const && !rhs_is_const && isCommutative()) {
+      auto constant_op = dyn_cast<neura::ConstantOp>(lhs.getDefiningOp());
 
-    Attribute rhs_const_value = getOriginConstantValue(rhs);
-    Operation *fused_op =
-        createOpWithFusedRhsConstant(op, rhs_const_value, rewriter);
+      Attribute lhs_const_value = getOriginConstantValue(lhs);
+      Operation *fused_op =
+          createOpWithFusedRhsConstant(op, rhs, lhs_const_value, rewriter);
 
-    rewriter.replaceOp(op, fused_op->getResults());
-    if (constant_op->use_empty()) {
-      rewriter.eraseOp(constant_op);
+      rewriter.replaceOp(op, fused_op->getResults());
+      if (constant_op->use_empty()) {
+        rewriter.eraseOp(constant_op);
+      }
+      return success();
     }
-    return success();
+
+    return failure();
   }
 };
 
 struct FuseAddRhsConstantPattern : public FuseRhsConstantPattern<neura::AddOp> {
   using FuseRhsConstantPattern<neura::AddOp>::FuseRhsConstantPattern;
 
+  bool isCommutative() const override { return true; }
+
   Operation *
-  createOpWithFusedRhsConstant(neura::AddOp op, Attribute rhs_const_value,
+  createOpWithFusedRhsConstant(neura::AddOp op, Value non_const_operand,
```

Comment:
Should we then rename `createOpWithFusedRhsConstant` to `createOpWithFusedConstant`